resolwe/flow/models/data.py

```diff
@@ -319,31 +322,55 @@ def create_entity(self):
         any `Entity`, create new `Entity`
 
         """
-        ds_slug = self.process.flow_collection  # pylint: disable=no-member
-        if ds_slug:
-            entity_query = Entity.objects.filter(data__in=self.parents.all()).distinct()  # pylint: disable=no-member
-
-            if entity_query.count() == 1:
-                entity = entity_query.first()
-
-                copy_permissions(entity, self)
+        entity_type = self.process.entity_type  # pylint: disable=no-member
+        entity_descriptor_schema = self.process.entity_descriptor_schema  # pylint: disable=no-member
+        entity_input = self.process.entity_input  # pylint: disable=no-member
+
+        if entity_type:
+            data_filter = {}
+            if entity_input:
+                input_id = dict_dot(self.input, entity_input, default=lambda: None)
+                if input_id is None:
+                    logger.warning("Skipping creation of entity due to missing input.")
+                    return
+                if isinstance(input_id, int):
+                    data_filter['data__pk'] = input_id
+                elif isinstance(input_id, list):
+                    data_filter['data__pk__in'] = input_id
+                else:
+                    raise ValueError("")
```

Comment:
You could also print out something like Skipping creation of entity due to invalid value of field {}. We should check the field in register that it is valid under process schema and that the field type is `data:*` or `list:data:*`. Bioinformaticians will make typos.